src/toolchain/toolchain.ts

```diff
@@ -212,14 +214,28 @@ export class SwiftToolchain {
                 if (!version) {
                     throw Error("Info.plist is missing the XCTEST_VERSION key.");
                 }
-                return path.join(
-                    platformPath,
-                    "Developer",
-                    "Library",
-                    `XCTest-${version}`,
-                    "usr",
-                    "bin"
-                );
+
+                let bindir: string;
+                const arch = targetInfo?.target?.triple.split("-", 1)[0];
+                switch (arch) {
+                    case "x86_64":
+                        bindir = "bin64";
+                        break;
+                    case "i686":
+                        bindir = "bin32";
+                        break;
+                    case "aarch64":
+                        bindir = "bin64a";
+                        break;
+                    default:
+                        throw Error(`unsupported architecture ${arch}`);
+                }
+
+                const root = path.join(platformPath, "Developer", "Library", `XCTest-${version}`);
+                if (existsSync(path.join(root, "usr", bindir))) {
```

Comment:
I see you are basing this on the existence of the folder, and not swift version. Is this your preferable option? You could pass in `swiftVersion` calculated in `SwiftToolchain.create` into this and add a test 
```ts
if (swiftVersion >= new Version(5,7,0)) {
...
}
```
I'm not too fussed either way. The only thing I would say is the version check is clearer in code.

Reply:
Also if you don't want to use the swift version, could you use an async call from `fs/promises` instead of the blocking call. There is no version of `exists` but you could use `stat` I guess.

Reply:
I think that your point has more merit than you ascribe to it.  Really, practically speaking, the benefits aren't worth it here.  The `stat` check gives you compatibility with 1 (maybe 2) snapshots of 5.7, which are broken for other reasons.  Really, the check is only marginally more useful in theory and practically speaking, the version check is both easier to reason about and cleaner.  I'd argue that changing the code here is better.